Modules/CalcPerform.lua

```diff
@@ -14,9 +14,308 @@ local m_ceil = math.ceil
 local m_floor = math.floor
 local m_modf = math.modf
 local s_format = string.format
+local bor = bit.bor
+local band = bit.band
 
 local tempTable1 = { }
 
+-- Identify the trigger action skill for trigger conditions, take highest Attack Per Second 
+local function findTriggerSkill(env, skill, source, triggerRate, reqManaCost)
+	local uuid = cacheSkillUUID(skill)
+	if not GlobalCache.cachedData["CACHE"][uuid] then
+		calcs.buildActiveSkill(env.build, "CACHE", skill)
+		env.dontCache = true
+	end
+
+	if GlobalCache.cachedData["CACHE"][uuid] then
+		-- Below code sets the trigger skill to highest APS skill it finds that meets all conditions
+		if not source and GlobalCache.cachedData["CACHE"][uuid].Speed then
```

Comment:
Don't all these if statements resolve to the same return value?

Reply:
source and triggerRate are input/output variables in this case. If the various "if" conditions are met they are updated with newer values, otherwise they are retained.

Reply:
Yeah, but I'm saying those if statements can all be combined into one, really.  Like the below (variables added for more readability)
```suggestion
		local cachedSpeed = GlobalCache.cachedData["CACHE"][uuid].Speed
		local cachedManaCost = GlobalCache.cachedData["CACHE"][uuid].ManaCost
		if ((not source and cachedSpeed)
			or (cachedSpeed and cachedSpeed > triggerRate))
			and ((reqManaCost and cachedManaCost and cachedManaCost >= reqManaCost)
			or not reqManaCost)
```

Reply:
You're right. Fixed.